Storage/Bookmarks/BookmarksModel.swift

```diff
@@ -296,6 +296,47 @@ open class PrependedBookmarkFolder: BookmarkFolder {
     }
 }
 
+open class ConcatenatedBookmarkFolder: BookmarkFolder {
+    fileprivate let main: BookmarkFolder
+    fileprivate let prepend: BookmarkFolder
+
+    init(main: BookmarkFolder, prepend: BookmarkFolder) {
+        self.main = main
+        self.prepend = prepend
+        super.init(guid: main.guid, title: main.guid)
+    }
+
+    override open var count: Int {
+        return main.count + prepend.count
+    }
+
+    override open subscript(index: Int) -> BookmarkNode? {
+        if index < prepend.count {
+            return prepend[index]
+        } else {
```

Comment:
On the topic of style: several of us dramatically prefer the Pythonic `self.foo` style when accessing member variables, because it avoids any chance of variable shadowing.